src/app/component/app-nav-bar/app-nav-bar.component.ts

```diff
@@ -339,7 +358,9 @@ export class AppNavBarComponent implements OnInit, OnChanges {
   }
 
   handleNavigateBack(): void {
-    window.history.back()
+    if (this.previousUrl.includes('/app/toc/do_') || this.previousUrl.includes('/viewer/pdf/do_')) {
+      this.router.navigateByUrl('/page/home')
```

Comment:
what is the logic here?
/viewer/pdf/do_ may be wrong condition if you have to check for viewer page in condition. This will work with only pdf resources not other types of resources